data-model/src/main/java/org/alfresco/repo/content/ContentStore.java

```diff
@@ -344,4 +345,38 @@ default DirectAccessUrl requestContentDirectUrl(String contentUrl, boolean attac
     {
         return Collections.emptyMap();
     }
+
+    /**
+     * Submit a request to send content to archive (offline) state.
+     * If no connector is present or connector is not supporting sending to archive, then {@code false} will be returned.
+     * Specific connector will decide which storage class/tier will be set for content.
+     * This method is experimental and subject to changes.
+     *
+     * @param contentUrl the URL of the content which is to be archived.
+     * @return true when request successful, false when unsuccessful.
+     */
+    @Experimental
+    default boolean requestSendContentToArchive(String contentUrl)
+    {
+        throw new UnsupportedOperationException("Request to archive content is not supported by this content store.");
+    }
+
+    /**
+     * Submit a request to restore content from archive (offline) state.
+     * If no connector is present or connector is not supporting restoring fom archive, then {@code false} will be returned.
+     * One of input parameters of this method is a map (String-Serializable) of Storage Provider specific input needed to perform proper restore.
+     * Keys of this map should be restricted to {@code ContentRestoreParams} enumeration.
+     * For AWS S3 map can indicating expiry days, Glacier restore tier.
+     * For Azure Blob map can indicate rehydrate priority.
+     * This method is experimental and subject to changes.
+     *
+     * @param contentUrl    the URL of the content which is to be archived.
+     * @param restoreParams a map of String-Serializable parameters defining Storage Provider specific request parameters (can be empty).
+     * @return true when request successful, false when unsuccessful.
+     */
+    @Experimental
+    default boolean requestRestoreContentFromArchive(String contentUrl, Map<String, Serializable> restoreParams)
```

Comment:
Should the `restoreParams` map be?:
```suggestion
    default boolean requestRestoreContentFromArchive(String contentUrl, Map<ContentRestoreParams, Serializable> restoreParams)
```

 I assume that was the intent for the enum? There is no String value for the ContentRestoreParams enum.

Reply:
Following call with @mpichura this can be resolved.
enum has a get value/ get name method baked in which can be used to make the map more generic and use String